### PR TITLE
Module Enumerable Methods feature

### PR DESCRIPTION
**Pull request by didymus707**
Contains The enumerable; my_each, my_each_with_index, my_all, my_none, my_any, my_select, my_map, my_count, my_inject, multiply_els method

### CODE REVIEW

**Comment by shshamim63** [comment]
## Status :red_circle: [Required Changes] :red_circle: 
__
Hi @didymus707 :wave: ,
Project contains a descriptive reamdme, code looks neat and clean, But the commit messages do not  look self descriptive to me. Actually the error fixing commit message. But for initial purpose great job :clap: . But there are some corners, needed to cover to get approval for this project. Lets identify those.
___
### :spades: [Mandatory Changes] :spades: 
___
1. [**`my_each`**](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)) 
...1. Returns an Enumerator if no block is given. But there is no check for that on your method.
2. [**`my_each_with_index`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. Returns an enumerator if no block is given. Nether can see the check here too.
3. [**`my_select`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. Returns an enumerator if no block is given.
4. [**`my_all?`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. When no block or argument is given, returns true when none of the collection members are false or nil. `[1, 2, 3].my_all?` should return true and `[1, 2, nil].my_all?` should return false.
...2. When a class is passed as an argument, returns true if all of the collection is a member of such class. `[1, 2, 3].my_all?(Integer)` should return true. else false.
...3.  when a Regex is passed as an argument, returns true if all of the collection matches the Regex. accept a regex and if that matches, should return true.
...4. When a pattern other than Regex or a Class is given, returns true if all of the collection matches the pattern. `[1, 1, 1].my_all?(1)` should return true.
5. [**`my_any?`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. when a class is passed as an argument, returns true if at least one of the collection is a member of such class `[1, 's','string'].my_any?(Integer)` should return true.
...2. when a Regex is passed as an argument, returns false if none of the collection matches the Regex. 
...3. when a pattern other than Regex or a Class is given, returns false if none of the collection matches the pattern. `[1, 2, 3].my_any?(1)` should return true.
6. [**`my_none?`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. when no block or argument is given, returns true only if none of the collection members is true.
`[1, 2 ,3].my_none?` should return false. `[false, false, false].my_none?` should return true.
...2. when a class is passed as an argument, returns true if none of the collection is a member of such class.
...3. when a Regex is passed as an argument, returns true only if none of the collection matches the Regex.
...4. when a pattern other than Regex or a Class is given, returns true only if none of the collection matches the pattern.
7. [**`my_map`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. Returns an Enumerator if no block is given.
8. [**`my_inject`**](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
...1. Combines each element of the collection by applying the symbol as a named method. So actually inject method can take 2 arguments. one is the initial value and another one is the symbol.
`[1, 2, 3].my_inject(1, :+)` should return 7.
___
Please fix the above mentioned issues and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Comment by didymus707** [comment]
But looking at odin and reading the docs some of these things you have added should be optional... Anyways, more work means more exercises to master the concepts

**Comment by didymus707** [comment]
Issues fixed

**Comment by Anna-Myzukina** [comment]
### Pull Request under the review by Anna Muzykina (TSE)

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:2]: Line number 2 gives an error "unexpected ..."

Comment (didymus707) [reply methods.rb:2]: It works when you comment it out. 
The purpose of the line 2 was to disable all cops

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:9]: Line number 9 kindly recommended to add space before for

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:17]: Line number 17 kindly recommended to add space before for

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:30]: Line number 30 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:57]: Line number 56 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb]: Line number 82 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:108]: Line number 98 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:126]: Line number 116 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb:139]: Line number 128 kindly recommended to add space before def

File: methods.rb
Comment (Anna-Myzukina) [inline methods.rb]: Line number 171 kindly recommended to fix syntax error end replace endc with end

**Comment by Anna-Myzukina** [comment]
### Your project looks :+1:  good!
### Great job thus far. Let`s fix enumerables my_all?, my_any? and my_none?  before it can be approved.


 * should have working my_all?  and returns the same thing as ruby's all?. You can read about  [all?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)

#### my_all? gives next error
any1

*  should have working my_any? and returns the same thing as ruby's any?. You can read about  [any?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F)

####  my_any? gives next error
any2

* should have working my_none?  and returns the same thing as ruby's none?.  You can read about  [none?](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F)

#### my_none? gives next errors
any3
none2
none3


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*

**Review by Anna-Myzukina — CHANGES_REQUESTED** [review]
The project still needs changes

**Comment by didymus707** [comment]
All methods completed
P.S Comment out line 2 to avoid an error when running the module

**Review by whiz25 — APPROVED** [review]
## Awesome :clap: @didymus707 
You have made all the required changes and therefore it's :+1: from me. Merge and continue.

### Martin Afani

**Pull request merged**
